l10n_th_tin_service/models/res_partner.py

```diff
@@ -0,0 +1,196 @@
+# Copyright 2020 Poonlap V. <poonlap@tanabutr.co.th>
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html)
+import logging
+import pprint
+import re
+from collections import OrderedDict
+from logging import DEBUG, INFO
+from os import path
+from pathlib import Path
+
+import requests
+from requests import Session
+from zeep import Client, Transport, helpers
+
+from odoo import _, api, models
+
+_logger = logging.getLogger(__name__)
+
+
+class ResPartner(models.Model):
+    _inherit = "res.partner"
+    tin_web_service_url = (
+        "https://rdws.rd.go.th/serviceRD3/checktinpinservice.asmx?wsdl"
+    )
+    vat_web_service_url = "https://rdws.rd.go.th/serviceRD3/vatserviceRD3.asmx?wsdl"
+
+    @staticmethod
+    def check_rd_tin_service(tin):
+        """Return bool after verifiying Tax Identification Number (TIN)
+           or Personal Identification Number (PIN)
+           by using Revenue Department's web service to prevent forging.
+           :param tin: a string for TIN or PIN
+        """
+
+        sess = Session()
+        mod_dir = path.dirname(path.realpath(__file__))
+        cert_path = str(Path(mod_dir).parents[0]) + "/static/cert/adhq1_ADHQ5.cer"
+        sess.verify = cert_path
+        transp = Transport(session=sess)
+        try:
+            cl = Client(ResPartner.tin_web_service_url, transport=transp)
+        except requests.exceptions.SSLError:
+            _logger.log(INFO, "Fall back to unverifed HTTPS request.")
+            sess.verify = False
+            transp = Transport(session=sess)
+            cl = Client(ResPartner.tin_web_service_url, transport=transp)
+        result = cl.service.ServiceTIN("anonymous", "anonymous", tin)
+        res_ord_dict = helpers.serialize_object(result)
+        _logger.log(DEBUG, pprint.pformat(res_ord_dict))
+        return res_ord_dict["vIsExist"] is not None
+
+    @staticmethod
+    def get_info_rd_vat_service(tin, branch=0):
```

Comment:
This should be private methods

```suggestion
    @staticmethod
    def _get_info_rd_vat_service(tin, branch=0):
```